x/gamm/types/hooks.go

```diff
@@ -3,7 +3,10 @@ package types
 import sdk "github.com/cosmos/cosmos-sdk/types"
 
 type GammHooks interface {
-	AfterPoolCreated(ctx sdk.Context, poolId uint64)
+	AfterPoolCreated(ctx sdk.Context, sender sdk.AccAddress, poolId uint64)
+	AfterJoinPool(ctx sdk.Context, sender sdk.AccAddress, poolId uint64, enterCoins sdk.Coins, shareOutAmount sdk.Int)
+	AfterExitPool(ctx sdk.Context, sender sdk.AccAddress, poolId uint64, shareInAmount sdk.Int, exitCoins sdk.Coins)
+	AfterSwap(ctx sdk.Context, sender sdk.AccAddress, poolId uint64, input sdk.Coins, output sdk.Coins)
```

Comment:
Can you add a comment to each of these specifying which messages will call it here as well?